proto/cosmos/app/v1alpha1/config.proto

```diff
@@ -0,0 +1,36 @@
+syntax = "proto3";
+
+package cosmos.app.v1alpha1;
+
+import "google/protobuf/any.proto";
+
+// Config represents the configuration for a Cosmos SDK ABCI app.
+// It is intended that all state machine logic including the version of
+// baseapp and tx handlers (and possibly even Tendermint) that an app needs
+// can be described in a config object. For compatibility, the framework should
+// allow a mixture of declarative and imperative app wiring, however, apps
+// that strive for the maximum ease of maintainability should be able to describe
+// their state machine with a config object alone.
+message Config {
+  // modules are the module configurations for the app.
+  repeated ModuleConfig modules = 1;
```

Comment:
Will modules order matter? Asking this question re: `SetOrder***` functions in simapp, and if it's related to this.

Reply:
If order doesn't matter, would it be useful to define some dependency graph between modules? (e.g. staking depends on bank). And this graph would deterministically define the `SetOrder***` orders.

Reply:
So set order would be an aspect of the App or BaseApp module which will receive begin blockers, etc from other modules as a map "module name -> begin blocker". This is done with one per module dependencies that we designed in container/